deepset_cloud_sdk/workflows/pipeline_client/pipeline_service.py

```diff
@@ -100,20 +125,128 @@ async def import_async(self, pipeline: PipelineProtocol, config: IndexConfig | P
+                await self._validate_index(config.name, pipeline_yaml)
+            else:
+                await self._validate_pipeline(config.name, pipeline_yaml)
+            logger.debug(f"Validation passed for {config.name}")
+        except DeepsetValidationError as err:
+            if config.strict_validation:
+                # Re-raise the error to fail the import
+                raise
+            # Log warning and continue with import
+            logger.warning(f"Validation issues found for {config.name}.")
+            logger.warning("Import will continue. Set strict_validation=True to fail on validation errors.")
+            for error_detail in err.errors:
+                logger.warning(f"Validation error [{error_detail.code}]: {error_detail.message}")
+
+    def _handle_validation_error(self, response: Response) -> None:
+        """Handle validation error response by extracting errors and raising DeepsetValidationError.
+
+        Supports multiple error response formats:
+        1. "details" field with code/message objects (preferred format)
+        2. "errors" field with objects containing "msg" and "type" fields
+        3. "errors" field with string values (fallback)
+        4. Non-JSON responses (fallback to raw text)
+
+        :param response: HTTP response object.
+        :raises DeepsetValidationError: Always raises with formatted error details.
+        """
+        response_json = (
+            response.json() if response.headers.get("content-type", "").startswith("application/json") else {}
+        )
+
+        if "details" in response_json:
+            error_details = [
+                ErrorDetail(code=detail["code"], message=detail["message"]) for detail in response_json["details"]
+            ]
+        elif "errors" in response_json and isinstance(response_json["errors"], list):
+            errors = response_json["errors"]
+            if errors and isinstance(errors[0], dict) and "msg" in errors[0] and "type" in errors[0]:
+                # Handle object-based errors with 'msg' and 'type' fields
+                error_details = [ErrorDetail(code=error["type"], message=error["msg"]) for error in errors]
+            else:
+                # Handle string-based errors
+                error_message = ", ".join(str(error) for error in errors)
+                error_details = [ErrorDetail(code=str(response.status_code), message=error_message)]
+        else:
+            error_details = [ErrorDetail(code="VALIDATION_FAILED", message=response.text)]
+
+        error_messages = []
+        for error in error_details:
+            error_messages.append(f"[{error.code}] {error.message}")
+            logger.error(f"Validation error [{error.code}]: {error.message}")
 
-    async def _import_index(self, pipeline: PipelineProtocol, config: IndexConfig) -> None:
+        raise DeepsetValidationError(
+            "Validation failed: " + "; ".join(error_messages), error_details, response.status_code
+        )
```

Comment:
Not 100% sure how the log lines would look like without that formatting and if its "acceptably ugly", but just printing the raw logs might be okay. Let's keep it like this. Just raw response bodies might have been okay here since we just wrap the api => let's focus on improving the API long terms